### PR TITLE
[MRG+1] Added zsh completion for Scrapy command-line tool and updated bash completion

### PR DESCRIPTION
**Pull request by Dineshs91**
Auto completion for zsh.

### CODE REVIEW

**Comment by nyov** [comment]
Will this see an update, or should it get closed?

**Comment by darkrho** [comment]
@Dineshs91 @nyov It works for me when copying the file to either ``/usr/share/zsh/vendor-functions/_scrapy_zsh_completion`` or ``/usr/share/zsh/vendor-completions/_scrapy``. Notice that in the latter case the filename must be ``_scrapy``.

**Comment by nyov** [comment]
Thanks for testing. it looks like I'm only having the `vendor-completions` folder on my system,
nothing else seems to have installed anything to `vendor-functions` or `site-functions`.
I would go with the latter version (also the filename of `_scrapy` instead of `_scrapy_zsh_completion`).
All together, `extras/scrapy_zsh_completion /usr/share/zsh/vendor-completions/_scrapy`

**Comment by Curita** [comment]
Can I suggest we add this as a plugin for [robbyrussell/oh-my-zsh](https://github.com/robbyrussell/oh-my-zsh) too? It's a pretty standardized collector of plugins for zsh. Creating a folder under `/plugins` named **scrapy** and adding the _scrapy_zsh_completion script there should be enough.

Not sure if this helps since it's not debian based, but in arch completions scripts are installed under `/usr/share/zsh/site-functions`.

**Comment by Dineshs91** [comment]
@darkrho Thanks for your inputs !

@nyov I have updated the pull request.

**Comment by Dineshs91** [comment]
@Curita Sure that's a great suggestion. I will create a PR later today.

It would be best, if the filename is _scrapy instead of _scrapy_zsh_completion. All plugins there, follow this convention. Completion file will be _pluginname.

arch completions scripts are installed under /usr/share/zsh/site-functions. This really helps. I am using a mac and this path (usr/share/zsh/vendor-completions/) doesn't exist, but /usr/share/zsh/site-functions is available.

File: scrapy_zsh_completion [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    case "$cmd" in
+        crawl|edit|check)
+            spiders=$(scrapy list 2>/dev/null) || spiders=""
+            if [[ -n "$spiders" ]]; then
+                compadd `echo $spiders`
+            fi
+        ;;
+        *)
+            if [[ CURRENT -eq 2 ]]; then
+                _arguments '*: :(check crawl deploy edit fetch genspider list parse runspider server settings shell startproject version view)'
```
Comment (kmike) [inline scrapy_zsh_completion]: we are removing 'scrapy deploy' command, I think it shouldn't be listed here

**Comment by Dineshs91** [comment]
Can anyone confirm if server command is available ?

Like ```$ scrapy server```

**Comment by kmike** [comment]
@Dineshs91 a good catch; `scrapy server` was removed a long time ago (it was a command for starting https://github.com/scrapy/scrapyd).

**Comment by Dineshs91** [comment]
@kmike Thanks for the prompt response. 

These commands should be removed from scrapy_bash_completion also.

**Comment by nyov** [comment]
@Dineshs91 
> This really helps. I am using a mac and this path (usr/share/zsh/vendor-completions/) doesn't exist, but /usr/share/zsh/site-functions is available.

Don't forget you're adding this to the `debian/` package path here, though. So for this PR, it should follow debian rules. For oh-my-zsh it's a different matter.

And I think no-one would mind if you fix scrapy_bash_completion as part of this ticket.

**Comment by Dineshs91** [comment]
1. In detail
 ```
$ scrapy [TAB]
scrapy commands
check         -- Check spider contracts,
crawl         -- Run a spider,
edit          -- Edit spider,
```

2. Just the commands
```
$ scrapy [TAB]

check crawl edit fetch ....
```

Which one is better ?

**Pull request merged**